### PR TITLE
test curl command + other UI improvements in the proxy config page

### PR DESCRIPTION
**Pull request by thomasmaas**
- [x] add curl
- [x] improve visual structure
 ~~make more clear when staging is out of date~~

wip:
Screenshot 2019-10-09 16 58 57

Screenshot 2019-10-17 16 13 06
Screenshot 2019-10-17 16 12 52
Screenshot 2019-10-17 16 11 35

### CODE REVIEW

File: app/views/api/integrations/apicast/configuration_driven/_apicast.html.slim [hunk 1]
```diff
@@ -50,13 +51,20 @@ section.Configuration class=('Configuration--is-promoted' unless deployment_opti
...
+          dl.u-dl
+            dt.u-dl-term URL
+            dd.u-dl-definition = @show_presenter.staging_proxy_endpoint
+            dt.u-dl-term Version
+            dd.u-dl-definition
+              ' v.
+              = @show_presenter.last_sandbox_config.version
+            dt.u-dl-term Test Curl
+            dd.u-dl-definition
+              = api_test_curl(@proxy)
```
Comment (guicassolato) [inline app/views/api/integrations/apicast/configuration_driven/_apicast.html.slim]: A couple enhancements to do here:
1. Probably render only if the product has applications, similar to https://github.com/3scale/porta/blob/22b1411b8eda4b4d1e54818d89db209f39c9dbf6/app/views/api/integrations/apicast/shared/_curl.html.slim#L4
2. Add the path for the first backend to the URI – requires refactoring of https://github.com/3scale/porta/blob/22b1411b8eda4b4d1e54818d89db209f39c9dbf6/app/helpers/api/integrations_helper.rb#L3

File: app/helpers/api/integrations_helper.rb [hunk 2]
```diff
@@ -1,36 +1,139 @@
...
-    case proxy.credentials_location
-    when 'headers'
-      credentials.each { |k, v| extheaders += " -H'#{k}: #{v}'" }
-    when 'query'
-      test_path = proxy.api_test_path
-      if test_path
-        query = "#{(test_path.index('?') ? '&' : '?')}#{credentials.to_query}"
-      else
-        query = "?#{credentials.to_query}"
+  class CurlCommandBuilder
```
Comment (guicassolato) [inline app/helpers/api/integrations_helper.rb]: This needs test.

File: app/helpers/api/integrations_helper.rb [hunk 3]
```diff
@@ -1,36 +1,139 @@
...
-      uri.user, uri.password = proxy.authorization_credentials
+      "curl \"#{uri.to_s}\" #{extheaders}"
+    end
+
+    def base_endpoint
+      raise NoMethodError, __method__
+    end
+
+    def path
+      apiap? ? proxy.service.backend_api_configs.first.path : proxy.api_test_path
```
Comment (guicassolato) [inline app/helpers/api/integrations_helper.rb]: I'll change this to build from the first mapping rule instead.

File: app/assets/stylesheets/provider/_forms.scss [hunk 4]
```diff
@@ -573,6 +573,8 @@ li.hidden {
 #integration-tabs {
   form.formtastic {
     box-shadow: none;
+    margin: 0;
```
[the remark this answers is not in the document]
Comment (damianpm) [reply app/assets/stylesheets/provider/_forms.scss]: Yes, it's OK, the style must be the same for both

[the remark this answers is not in the document]
Comment (damianpm) [reply app/assets/stylesheets/provider/_forms.scss]: For example
**APIAP:**
apiap

No **APIAP:**
no-apiap

[the remark this answers is not in the document]
Comment (damianpm) [reply app/assets/stylesheets/provider/_forms.scss]: Hmm, wait a minute, I've found a form affected in the wrong way, fixing now

[the remark this answers is not in the document]
Comment (damianpm) [reply app/assets/stylesheets/provider/_forms.scss]: Fixed, adding screenshots in Conversation page

**Comment by damianpm** [comment]
Fixed Alignment of "Promote" forms

APIAP:

> apiap

No APIAP:

> no-apiap


NO APIAP Integration Page:

> no-apiap-integration-page

@thomasmaas @guicassolato

**Pull request merged**